### PR TITLE
drivers: several cleanups related to xtimer + some ztimer migration

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR fixes several minor issues with xtimer in some drivers and migrate them to ztimer (except the ws281x test app): in some cases, xtimer dependency was pulled in in the driver while not used there (only in the test app), in come other cases it was the opposite (xtimer pulled-in from the test app while it's a dependency of the driver), there was also a remaining unused xtimer include.

### Testing procedure

- Green CI
- We could test the test applications related to the affected drivers but I don't have access to this hardware.

### CODE REVIEW

File: tests/driver_lis3mdl/main.c [hunk 1]
```diff
@@ -21,11 +21,12 @@
...
 #include <stdio.h>
 
-#include "xtimer.h"
+#include "timex.h"
+#include "ztimer.h"
 #include "lis3mdl.h"
 #include "lis3mdl_params.h"
 
-#define SLEEP_USEC  (800 * 800U)
+#define SLEEP_MS  (640 * MS_PER_SEC)
```
Comment (maribu) [inline tests/driver_lis3mdl/main.c:29]:
```suggestion
#define SLEEP_MS  (640)
```

Otherwise the duration would change

**Review by maribu — COMMENTED** [review]
lgtm, I think I found one conversion issue (see inline)

File: tests/driver_lis3mdl/main.c (hunk 1 above)
Comment (aabadie) [reply tests/driver_lis3mdl/main.c:29]: Thanks! Stupid mistake...

**Review by maribu — APPROVED** [review]

**Comment by aabadie** [comment]
Thanks!

**Pull request merged**